src/diffusers/configuration_utils.py

```diff
@@ -109,13 +109,6 @@ def register_to_config(self, **kwargs):
         # TODO: remove this when we remove the deprecation warning, and the `kwargs` argument,
         # or solve in a more general way.
         kwargs.pop("kwargs", None)
-        for key, value in kwargs.items():
-            try:
-                setattr(self, key, value)
```

Comment:
I think we should delete this, I don't know why I added it back then, but it's not a good design looking at it now.

In short currently, we have the following logic:

All diffusers models and pipelines shall inherit from `ConfigMixin` which gives some nice config utils:
- the `.config` property`
- automatic saving of arguments pass to the class contstructor
- load / save config 
- from_config

The API is designed to allow the following
```
class MyModel(ConfigMixin):

    @register_to_config
    def __init__(self, attr1, attr2):
        super().__init__()
```

This currently does two things:

a) it automatically saves `attr1` and `attr2` as `self.config.attr1` and `self.config.attr2`
b) **However**, it also saves `attr1` and `attr2` as `self.attr1` and `self.attr2`

I'm not sure why I thought b) is a good idea back then, but I think it's a pretty bad design choice now as it's not intuitive and also prone to errors (see raise error statement here). Also it doesn't really help as every attribute can be accessed with `self.config` . Also it badly entangles object members such as `pipe.unet` with configs which are not models.

=> Long story short, I think we should remove option **b)**. This will clearly be a breaking change, but I don't think anybody really used b) much before as it's quite weird to assume that config attributes are direct members of the class instance.


@pcuenca @sayakpaul @patil-suraj @williamberman @yiyixuxu ok for you if I remove this with a big 🚨 headear? 

As you can see many tests are failing, but they are all easy to fix.

Reply:
+100 to your reasoning. I am okay with this.

Reply:
This way, we also encourage our users to access configuration variables from `.config` which reduces the cognitive burdens and helps establish a clean and uniform design philosophy.

Reply:
yep!

Reply:
An alternative to minimize breaking changes would be to implement `__getattr__` and return the property if it's in the `config`. It's not so strange to think that, i.e., `in_channels` could be an attribute of `unet`.

Reply:
This changes broke [this pipeline](https://github.com/mikonvergence/ControlNetInpaint)

I think it could break many more, instead of not setting attributes anymore we should set getters that tell the user these properties are deprecated, should i open a PR?

Reply:
That's fair. I think it should be nice. Cc @patrickvonplaten

Reply:
This PR should work for the patch release: https://github.com/huggingface/diffusers/pull/3129

Reply:
@remorses sorry about the breaking change, we just made a patch release for it